### PR TITLE
Sorting in salons_controller

### PR DESCRIPTION
**Pull request by Vgoylo**
Add sorting in controller to salons

### CODE REVIEW

File: app/controllers/api/v1/salons_controller.rb [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
-      before_action :set_salon, only: %i[show update destroy] 
+      before_action :set_salon, only: %i[show update destroy]
 
       attr_accessor :salon
 
       def index
-        @salons = Salon.all
-        render json: @salons
+
+        if params[:sort]
```
Comment (TehanovAnton) [inline app/controllers/api/v1/salons_controller.rb:10]: look my comments in #11

File: app/controllers/api/v1/salons_controller.rb [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
     class SalonsController < ApplicationController
-      before_action :set_salon, only: %i[show update destroy] 
+      before_action :set_salon, only: %i[show update destroy]
 
       attr_accessor :salon
 
       def index
-        @salons = Salon.all
-        render json: @salons
+        render json: params[:sort] ? Salon.order(sort_params.to_h) : Salon.all
```
Comment (dmokreckiy) [inline app/controllers/api/v1/salons_controller.rb:9]: please separate logic from ternary operator and extract it to normal variables to improve readability.

File: app/controllers/api/v1/salons_controller.rb [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+      end
+
+      def sort_params
+        params.require(:sort).permit!
```
Comment (dmokreckiy) [inline app/controllers/api/v1/salons_controller.rb:13]: `.permit!`? Why use strong params at all then?
Please use strong params normally.

**Review by dmokreckiy — CHANGES_REQUESTED** [review]
Please refactor logic and strong params

**Review by GEK62 — APPROVED** [review]

**Pull request merged**